src/nnet3/nnet-test-utils.cc

```diff
@@ -888,13 +888,21 @@ void ComputeExampleComputationRequestSimple(
   request->outputs.clear();
   inputs->clear();
 
+  int32 t_modulus = GetTimePeriodForIvector(nnet);
   std::vector<Index> input_indexes, ivector_indexes, output_indexes;
   for (int32 n = n_offset; n < n_offset + num_examples; n++) {
     for (int32 t = input_start_frame; t < input_end_frame; t++)
       input_indexes.push_back(Index(n, t, 0));
     for (int32 t = output_start_frame; t < output_end_frame; t++)
       output_indexes.push_back(Index(n, t, 0));
-    ivector_indexes.push_back(Index(n, 0, 0));
+    if (t_modulus == 0) // case 1: single ivector for the entire chunk
```

Comment:
Aah yes, I did not consider the case of multiple-chunks. In that case it is makes sense to rely on the GetTimePeriod* function's output. 

@freewym In this case just changing the name of the variable is sufficient. Please ignore my comments on single/multiple ivector case and address the other issues.